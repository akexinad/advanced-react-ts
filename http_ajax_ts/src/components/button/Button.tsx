import React, { FC } from "react";

import styles from "./Button.module.css";

interface ButtonProps {
    label: string;
}

const Button: FC<ButtonProps> = ({ label }) => {
    return <div data-testid="button" className={styles.Button}>{label}</div>;
};

export default Button;
