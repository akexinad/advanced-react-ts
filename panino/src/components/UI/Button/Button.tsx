import React, { FC } from "react";

import styles from "./Button.module.css";

interface ButtonProps {
    clicked: () => void;
    btnType: "Danger" | "Success";
}

const Button: FC<ButtonProps> = ({ children, clicked, btnType }) => (
    <button
        className={[styles.Button, styles[btnType]].join(" ")}
        onClick={clicked}
    >
        {children}
    </button>
);

export default Button;
