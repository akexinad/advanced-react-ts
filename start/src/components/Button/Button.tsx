import React, { FC } from 'react'

interface MockProps {
    foo: string
    bar: number
    baz: boolean
}

interface ButtonClickProps {
    buttonClick: MockProps[]
}

const Button: FC<ButtonClickProps> = ({ buttonClick }) => {

    const clickHandler = () => {
        console.log(buttonClick);
        
        console.log("clicked!");
    }

    return  (
        <button onClick={clickHandler}>
            Btn Component
        </button>
    )

}

export default Button;