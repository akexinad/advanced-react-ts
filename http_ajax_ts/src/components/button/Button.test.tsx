import React from "react";
import ReactDOM from "react-dom";

import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

import Button from "./Button";

const buttonLabel = "click me";
const buttonComponent = <Button label={buttonLabel} />;

afterEach(cleanup);

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(buttonComponent, div);
});

it("renders correctly", () => {
    const { getByTestId } = render(buttonComponent);
    getByTestId("button");

    // This will pass because it matches the snapshot.
    expect(getByTestId("button")).toHaveTextContent(buttonLabel + "FOO");

    // This will fail because the label has been changed.
    expect(getByTestId("button")).toHaveTextContent(buttonLabel);
});


/**
 * Why are snapshots useful?
 * 
 * When you commit the code, the test suite will be able to check committed snapshots when another engineer makes a change.
 */

it("matches the snapshot", () => {
    const tree = renderer.create(buttonComponent).toJSON();
    expect(tree).toMatchSnapshot();
});
