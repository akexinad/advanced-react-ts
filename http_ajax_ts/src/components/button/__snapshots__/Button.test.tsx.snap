// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="Button"
  data-testid="button"
>
  click me
</div>
`;
